completions/make

```diff
@@ -157,19 +157,65 @@ _make()
             fi
         done
 
-        # recognise that possible completions are only going to be displayed
-        # so only the base name is shown
+        # recognise that possible completions are only going to be displayed so
+        # only the base name is shown.
+        #
+        # Note: This is currently turned off because the test suite of
+        # bash-completion conflicts with it; it uses "set show-all-if-ambiguous
+        # on" (causing COMP_TYPE == 37) to retrieve the action completion
+        # results, and also the compact form with only the basenames is not
+        # essentially needed.  To re-enable it, please uncomment the following
+        # if-statement.
         local mode=--
-        if ((COMP_TYPE != 9)); then
-            mode=-d # display-only mode
-        fi
+        # if ((COMP_TYPE != 9 && COMP_TYPE != 37 && COMP_TYPE != 42)); then
+        #     mode=-d # display-only mode
+        # fi
 
         local IFS=$' \t\n' script=$(_make_target_extract_script $mode "$cur")
         COMPREPLY=($(LC_ALL=C \
             $1 -npq __BASH_MAKE_COMPLETION__=1 \
             ${makef+"${makef[@]}"} "${makef_dir[@]}" .DEFAULT 2>/dev/null |
             command sed -ne "$script"))
 
+        local prefix=$cur
```

Comment:
Could we add some commentary here what the added hunk of code does, and why? I may be even slower today than usual, but it's not clear to me after eyeballing it for a while, let alone if revisiting afterwards without the context in the PR available to crosscheck.

Reply:
Thank you for the suggestion. I have added necessary code comments in commit 0ef11ca15b98abb47394f7c39208812239a6c379 (which was later squashed). I also noticed that this implementation does not work expected, so I modified the code too. The corresponding tests are added in commit 5df3eef90babd6b4553fa5c011bd1c3cd074d4dd (which was also squashed).